NewForce-Cohort-5/tabloidfullstack-fool-s-spring
# Tags - Add Tag to A Post

# Description

This is a new feature that allows user add a tag to the user's post. A user logs in, then navigates to their post, using the My Post link in the navigation bar. A list of posts associated with the current logged in user should be displayed. The user then clicks on the **Title** of the post they wish to add tags too, where they are then redirected to Post Details. On the Post Details page, a **Manage Tags** button is shown. User clicks on that button, where a list of available tags are shown in a list format with checkboxes. User can click which tags they wish to add then hit **Save**, after which user is then redirected back to the Post Details page and tags are viewable at the bottom. 

### [TICKET #17 Add a Tag to a Post](https://trello.com/c/Qsd2bDRv/17-17-add-a-tag-to-a-post)

_Note: Delete Tag From Post is a seperate ticket #18_

## Type of change


# Testing Instructions

- run `git add .` then `git commit` all changes in your terminal
- `git checkout main` then `git fetch -a` to get all branches that haven't been fetched yet
- `git checkout ajb-tag-edit`
- `start ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid.sln`
- Run these queries in your SQL Object Explorer [Create DB](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/01_Tabloid_Create_DB.sql) and [Seed Data](https://github.com/NewForce-Cohort-5/tabloidfullstack-fool-s-spring/blob/main/SQL/02_Tabloid_Seed_Data.sql)
- In Visual Studio, run in debugger using the `Tabloid` server
- `cd ~{working-directory}/tabloidfullstack-fool-s-spring/Tabloid/client && npm start`
- Log in as 'foo@bar.com'
- Click the My Post link in the navigation bar
- You should now see a list of all posts for the foo@bar.com logged in user
- Click on the Title of the Post you wish to add tags to
- You should be redirected to the Post Details
- At the bottom of post, click on "Manage Tags" button
- In the Add Tags form, a list of available tags is shown
- Click the checkbox next to the tag you wish to add
- Note that tags that are already added to the post should already be pre-checked in the list
- After choosing which tags to add, click "Save" button
- You should be redirect to to the Post Details page, where your tags should show up at the bottom of the post


# Checklist:
